qcodes/instrument/parameter.py

```diff
@@ -1292,33 +1305,60 @@ def _update_with(self, *,
         def __call__(self) -> ParamDataType:
             return self.get(get_if_invalid=True)
 
-    def __init__(self, name: str, source: Parameter, *args: Any,
+    def __init__(self, name: str, source: Optional[Parameter], *args: Any,
                  **kwargs: Any):
-        self.source = source
 
-        for ka, param in zip(('unit', 'label', 'snapshot_value'),
-                             ('unit', 'label', '_snapshot_value')):
-            kwargs[ka] = kwargs.get(ka, getattr(self.source, param))
+        for attr in ('unit', 'label'):
+            kwargs[attr] = kwargs.get(attr, getattr(source, attr, None))
 
         for cmd in ('set_cmd', 'get_cmd'):
             if cmd in kwargs:
                 raise KeyError(f'It is not allowed to set "{cmd}" of a '
                                f'DelegateParameter because the one of the '
                                f'source parameter is supposed to be used.')
-
+        self.source = source
         super().__init__(name, *args, **kwargs)
+        # explicitly set the source properties as
+        # init will overwrite the ones set when assigning source
+        self._set_properties_from_source(source)
+
         delegate_cache = self._DelegateCache(self)
         self.cache = cast(_Cache, delegate_cache)
 
-    # Disable the warnings until MultiParameter has been
-    # replaced and name/label/unit can live in _BaseParameter
+    @property
+    def source(self) -> Optional[Parameter]:
+        return self._source
+
+    @source.setter
+    def source(self, source: Optional[Parameter]) -> None:
+        self._set_properties_from_source(source)
+        self._source: Optional[Parameter] = source
+
+    def _set_properties_from_source(self, source: Optional[Parameter]) -> None:
+        if source is None:
+            self.gettable = False
+            self.settable = False
+            self._snapshot_get = False
+        else:
+            self.gettable = source.gettable
+            self.settable = source.settable
+            self._snapshot_get = source._snapshot_get
```

Comment:
Is this what we want? Should the parameter always follow its source or should we allow this to be overwritten in the Delgate

Reply:
>Should the parameter always follow its source

yes, as discused offline